cli/tests/pcluster/api/controllers/test_cluster_operations_controller.py

```diff
@@ -1033,29 +1034,411 @@ def test_aws_api_errors(self, client, mocker):
 
 
 class TestUpdateCluster:
-    def test_update_cluster(self, client):
-        update_cluster_request_content = {"clusterConfiguration": "clusterConfiguration"}
-        query_string = [
-            ("suppressValidators", "ALL"),
-            ["validationFailureLevel", ValidationLevel.INFO],
-            ("region", "eu-west-1"),
-            ("dryrun", True),
-            ("forceUpdate", True),
-            ("clientToken", "client_token_example"),
-        ]
+    url = "/v3/clusters/{cluster_name}"
+    method = "PUT"
+
+    BASE64_ENCODED_CONFIG = "SW1hZ2U6CiAgT3M6IGFsaW51eDIKSGVhZE5vZGU6CiAgSW5zdGFuY2VUeXBlOiB0Mi5taWNybw=="
+
+    def _send_test_request(
+        self,
+        client,
+        cluster_name,
+        region="us-east-1",
+        update_cluster_request_content=None,
+        suppress_validators=None,
+        validation_failure_level=None,
+        dryrun=None,
+        force_update=None,
+    ):
+        query_string = []
+        if region:
+            query_string.append(("region", region))
+        if suppress_validators:
+            query_string.extend([("suppressValidators", validator) for validator in suppress_validators])
+        if validation_failure_level:
+            query_string.append(("validationFailureLevel", validation_failure_level))
+        if dryrun is not None:
+            query_string.append(("dryrun", dryrun))
+        if force_update is not None:
+            query_string.append(("forceUpdate", force_update))
+
         headers = {
             "Accept": "application/json",
             "Content-Type": "application/json",
         }
-        response = client.open(
-            "/v3/clusters/{cluster_name}".format(cluster_name="clustername"),
-            method="PUT",
+        return client.open(
+            self.url.format(cluster_name=cluster_name),
+            method=self.method,
             headers=headers,
-            data=json.dumps(update_cluster_request_content),
-            content_type="application/json",
             query_string=query_string,
+            data=json.dumps(update_cluster_request_content) if update_cluster_request_content else None,
+        )
+
+    @pytest.mark.parametrize(
+        "update_cluster_request_content, suppress_validators, validation_failure_level, force_update",
+        [
+            (
+                {
+                    "clusterConfiguration": BASE64_ENCODED_CONFIG,
+                },
+                None,
+                None,
+                None,
+            ),
+            (
+                {
+                    "clusterConfiguration": BASE64_ENCODED_CONFIG,
+                },
+                ["type:type1", "type:type2"],
+                ValidationLevel.WARNING,
+                False,
+            ),
+        ],
+        ids=["required", "all"],
+    )
+    def test_successful_update_request(
```

Comment:
Shall we test update with no config changes? what happens in this case?

Reply:
In this case the validation succeeds. The current PCluster 3 behavior is that the update is triggered and then the CLI notifies that there were no updates (see [code](https://github.com/aws/aws-parallelcluster/blob/wip/pcluster3/cli/src/pcluster/cli_commands/update.py#L138))

Reply:
As agreed offline, we will throw a BadRequestException in case of an empty changeset.